addons/sale_project/models/project_milestone.py

```diff
@@ -7,18 +7,37 @@ class ProjectMilestone(models.Model):
     _name = 'project.milestone'
     _inherit = 'project.milestone'
 
+    def _default_sale_line_id(self):
+        project_id = self._context.get('default_project_id')
+        if not project_id:
+            return []
+        project = self.env['project.project'].browse(project_id)
+        return project.sale_line_id.filtered(lambda sol: sol.qty_delivered_method == 'milestones')
+
     allow_billable = fields.Boolean(related='project_id.allow_billable')
     project_partner_id = fields.Many2one(related='project_id.partner_id')
 
-    sale_line_id = fields.Many2one('sale.order.line', 'Sales Order Item', help='Sales Order Item that will be updated once the milestone is reached.',
+    sale_line_id = fields.Many2one('sale.order.line', 'Sales Order Item', default=_default_sale_line_id, help='Sales Order Item that will be updated once the milestone is reached.',
         domain="[('order_partner_id', '=?', project_partner_id), ('qty_delivered_method', '=', 'milestones')]")
-    quantity_percentage = fields.Float('Quantity', help='Percentage of the ordered quantity that will automatically be delivered once the milestone is reached.')
+    quantity_percentage = fields.Float('Quantity Percentage', compute="_compute_quantity_percentage", store=True, help='Percentage of the ordered quantity that will automatically be delivered once the milestone is reached.')
```

Comment:
```suggestion
    quantity_percentage = fields.Float('Quantity (%)', compute="_compute_quantity_percentage", store=True, help='Percentage of the ordered quantity that will automatically be delivered once the milestone is reached.')
```
So that you don't have to precise it in the view?